Library/Homebrew/utils.rb

```diff
@@ -10,102 +10,28 @@
 require "utils/analytics"
 require "utils/github"
 require "utils/curl"
-
-class Tty
-  class << self
-    def strip_ansi(string)
-      string.gsub(/\033\[\d+(;\d+)*m/, "")
-    end
-
-    def blue
-      bold 34
-    end
-
-    def white
-      bold 39
-    end
-
-    def magenta
-      bold 35
-    end
-
-    def red
-      underline 31
-    end
-
-    def yellow
-      underline 33
-    end
-
-    def reset
-      escape 0
-    end
-
-    def em
-      underline 39
-    end
-
-    def green
-      bold 32
-    end
-
-    def gray
-      bold 30
-    end
-
-    def highlight
-      bold 39
-    end
-
-    def width
-      `/usr/bin/tput cols`.strip.to_i
-    end
-
-    def truncate(str)
-      w = width
-      w > 10 ? str.to_s[0, w - 4] : str
-    end
-
-    private
-
-    def color(n)
-      escape "0;#{n}"
-    end
-
-    def bold(n)
-      escape "1;#{n}"
-    end
-
-    def underline(n)
-      escape "4;#{n}"
-    end
-
-    def escape(n)
-      "\033[#{n}m" if $stdout.tty?
-    end
-  end
-end
+require "utils/tty"
 
 def ohai(title, *sput)
   title = Tty.truncate(title) if $stdout.tty? && !ARGV.verbose?
-  puts "#{Tty.blue}==>#{Tty.white} #{title}#{Tty.reset}"
+  puts "#{Tty.blue}==>#{Tty.reset} #{Tty.bold}#{title}#{Tty.reset}"
   puts sput
 end
 
 def oh1(title, options = {})
   if $stdout.tty? && !ARGV.verbose? && options.fetch(:truncate, :auto) == :auto
     title = Tty.truncate(title)
   end
-  puts "#{Tty.green}==>#{Tty.white} #{title}#{Tty.reset}"
+  puts "#{Tty.green}==>#{Tty.reset} #{Tty.bold}#{title}#{Tty.reset}"
 end
 
 # Print a warning (do this rarely)
 def opoo(warning)
-  $stderr.puts "#{Tty.yellow}Warning#{Tty.reset}: #{warning}"
+  $stderr.puts "#{Tty.yellow.underline}Warnin#{Tty.reset.yellow}g:#{Tty.reset} #{warning}"
```

Comment:
> Because the underline would collide with the `g`. This is really a minor detail. 😄

I don't understand this. Would you mind explaining it in a little more detail? Thanks!